polaris/utils/errors.py

```diff
@@ -10,12 +13,12 @@ class InvalidResultError(ValueError):
     pass
 
 
-class PolarisChecksumError(ValueError):
-    pass
-
+class PolarisHubError(Exception, FormattingMixin):
+    def __str__(self):
+        response_message = self.__cause__.response.text
```

Comment:
**Suggestion:** This feels brittle. There are no guarantees that the cause of the `PolarisHubError` has the `response` field. If you would like to show the `response.text` at the bottom of the stack trace, I would change it back such that you explicitly pass in the `response`. 

Maybe move it to the `PolarisHubError`? I think it makes sense to always require a `response` for these errors! 

```py
class PolarisHubError(Exception, FormattingMixin):
    def __init__(self, message: str, response: Response):
        super().__init__(message)
        self.response = response
```

Reply:
Did I misinterpret what you meant in these previous comments? That's why I went down this path.


<img width="807" alt="Screenshot 2024-07-22 at 7 45 15 AM" src="https://github.com/user-attachments/assets/49a16f54-fce5-41a7-bb0a-9d7e267b8dc3">

Reply:
My bad! I was thinking that by raising from another error, we could drop the `response.text` from the message in the error we throw (reason being that can be found in the stack trace now). 

However, thinking about this again, I prefer us adding the `response` parameter back to the `PolarisHubError` as you had it earlier. 
- Having actionable feedback at the end of the stack trace, rather than hiding it in the stack trace, is way more user friendly!
- I also like that this is adding value to the custom class we're introducing (e.g. we could now also more easily reformat the Zod errors in the JSON body) and that it makes the expected context in which you can throw such an error explicit, i.e. a `PolarisHubError` should only be thrown in the context of a request and response in the client!